core/rpc/core/src/impl/utils.rs

```diff
@@ -489,141 +408,126 @@ impl<C: CkbRpc> MercuryRpcImpl<C> {
                     .map_err(|e| CoreError::DBError(e.to_string()))?
             }
 
-            Item::Record(id) => {
-                let (outpoint, _ownership) = decode_record_id(id)?;
-                self.storage
-                    .get_transactions(
-                        ctx.clone(),
-                        Some(outpoint),
-                        vec![],
-                        type_hashes,
-                        range,
-                        limit_cellbase,
-                        pagination,
-                    )
-                    .await
-                    .map_err(|e| CoreError::DBError(e.to_string()))?
-            }
+            Item::OutPoint(out_point) => self
+                .storage
+                .get_transactions(
+                    ctx.clone(),
+                    Some(out_point.into()),
+                    vec![],
+                    type_hashes,
+                    range,
+                    limit_cellbase,
+                    pagination,
+                )
+                .await
+                .map_err(|e| CoreError::DBError(e.to_string()))?,
         };
 
         Ok(ret)
     }
 
-    pub(crate) fn get_secp_lock_hash_by_item(&self, item: Item) -> InnerResult<H160> {
+    pub(crate) fn get_secp_lock_hash_by_pubkey_hash(&self, pubkey_hash: H160) -> InnerResult<H160> {
+        let lock_hash: H256 = self
+            .get_script_builder(SECP256K1)?
+            .args(Bytes::from(pubkey_hash.0.to_vec()).pack())
```

Comment:
there is a default Pack implementation for `[u8]`

```
impl Pack<packed::Bytes> for [u8]
```

```suggestion
            .args(pubkey_hash.0.pack())
```